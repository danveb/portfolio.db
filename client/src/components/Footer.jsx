import { Link } from "react-router-dom";
import { socials } from "../constants/socials";
import "../styles/Footer.scss"; 

export default function Footer() {
    return (
        <div className="footer">
            <div className="footer__wrapper">
                <div className="footer__main">
                    {/* <div className="footer__head">
                        <h1>Social Links<span>.</span></h1>
                    </div> */}
                    <div className="footer__copyright">
                        <p>© All rights reserved – Danny Bae</p>
                    </div>
                    <div className="footer__social__icons">
                        <ul>
                            {socials.map((social) => (
                                <li key={social.id}>
                                    <Link to={social.url} target={social.target} rel={social.rel}>
                                        <img src={social.img} alt={social.alt} />
                                    </Link>
                                </li>
                            ))}
                        </ul>
                    </div>
                </div>
            </div>
        </div>
    )
}
