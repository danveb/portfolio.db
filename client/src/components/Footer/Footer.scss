.footer {
    margin: 0 30px 30px 30px; 
    background-color: var(--rich-black); 
    height: calc(100vh - 50vh - 50px); 

    .footer-wrapper {
        height: 100%; 
        padding: 20px; 
        display: flex; 
        justify-content: space-between; 
        flex-direction: column; 
        align-items: flex-start;

        .footer-headline {
            
            h1 {
                color: var(--alice-blue); 
                font-size: 36px; 
                margin: 10px 0;

                span {
                    color: var(--bottle-green); 
                }

                @media screen and (max-width: 768px) {
                    font-size: 30px; 
                }
                
                @media screen and (max-width: 480px) {
                    font-size: 20px; 
                }
            }
        }

        .footer-social {
            width: 100%; 

            li {
                padding: 10px 0; 

                @media screen and (max-width: 480px) {
                    padding: 13px 0; 
                }
                
                a {
                    color: var(--alice-blue); 
                    
                    svg {
                        height: 25px; 
                        width: 25px; 

                        @media screen and (max-width: 480px) {
                            height: 15px; 
                            width: 15px; 
                        }
                    }
                }
            }   
        }

        .footer-mr {
            p {
                font-size: 10px; 
                color: var(--cadet-blue-crayola); 
            }
        }
    }
}