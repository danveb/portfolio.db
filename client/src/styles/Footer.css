.footer {
  bottom: 0; 
  height: 50px; 
  width: 100%; 
}

.footer__wrapper {
  max-width: 768px; 
  margin: 0 auto; 
  width: 100%; 
  height: 100%; 
  background-color: var(--rich-black); 
}

.footer__main {
  display: flex; 
  justify-content: space-between; 
  align-items: center; 
  height: 100%; 
}

.footer__copyright p {
  color: #777; 
  font-family: inherit; 
  font-size: 11px; 
}

.footer__social__icons ul {
  display: flex; 
  justify-content: space-between; 
  gap: 20px; 
}

.footer__social__icons li {
  list-style: none; 
}

.footer__social__icons a img {
  width: 25px; 
  height: 25px; 
  object-fit: contain; 
  cursor: pointer; 
}

@media screen and (max-width: 1024px) {
  .footer__wrapper {
    padding: 0 20px; 
  }
}