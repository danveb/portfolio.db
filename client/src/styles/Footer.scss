.footer {
    height: 50px; 
    width: 100%; 
    background-color: var(--raisin-black); 

    .footer__wrapper {
        max-width: 1200px; 
        height: 100%; 
        width: 100%; 
        margin: 0 auto; 

        // .footer__head {
        //     color: var(--alice-blue); 
            
        //     h1 {
        //         font-size: 30px; 

        //         span {
        //             color: var(--bottle-green)
        //         }
        //     }
        // }

        .footer__main {
            display: flex; 
            justify-content: space-around; 
            align-items: center; 
            height: 100%; 

            .footer__copyright {
                color: #777; 

                p {
                    font-size: 11px; 
                }
            }

            .footer__social__icons {
                ul {
                    display: flex; 
                    justify-content: space-between; 
                    gap: 20px; 

                    li {
                        list-style: none; 
                        a {
                            img {
                                color: var(--alice-blue); 
                                width: 25px; 
                                height: 25px; 
                                cursor: pointer; 
                            }
                        }


                    }
                }
                
            }
        }
    }
}