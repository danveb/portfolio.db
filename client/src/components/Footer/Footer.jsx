import { socials } from "../../constants/socials"; 
import "./Footer.scss"; 

const Footer = () => {
    return (
        <div className="footer">
            <div className="footer-wrapper">
                <div className="footer-headline">
                    <h1>Social Links<span>.</span></h1>
                </div>
                <div className="footer-social">
                    <ul>
                        {socials.map((social) => (
                            <li key={social.id}>
                                <a href={social.url} target={social.target} rel={social.rel}>{<social.icon />}</a>
                            </li>
                        ))}
                    </ul>
                </div>
                <div className="footer-mr">
                    <p>Portfolio design & development - Danny Bae</p>
                </div>
            </div>
        </div>
    )
}

export default Footer