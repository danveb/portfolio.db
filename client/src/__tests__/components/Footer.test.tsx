import { render, screen } from "@testing-library/react"; 
import Footer from "../../components/Footer"; 
import { BrowserRouter } from "react-router-dom";

describe("Footer component", () => {
  // smoke test
  test("renders component wtesthout crashing", () => {
    render(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>
    ); 
  });

  // snapshot test 
  test("matches snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>
    ); 
    expect(asFragment()).toMatchSnapshot(); 
  }); 

  // getByAlt
  test("should display total 3 social icons alt text", () => {
    render(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>
    );
    const altText = screen.getAllByAltText("social icon"); 
    // console.log(altText); // ARRAY 
    expect(altText.length).toBe(3); 
    expect(altText.length).not.toBe(100); 
  });
})