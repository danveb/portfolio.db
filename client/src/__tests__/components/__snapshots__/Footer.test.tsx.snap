// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    <div
      class="footer__wrapper"
    >
      <div
        class="footer__main"
      >
        <div
          class="footer__copyright"
        >
          <p>
            © All rights reserved – Danny Bae
          </p>
        </div>
        <div
          class="footer__social__icons"
        >
          <ul>
            <li>
              <a
                href="https://www.github.com/danveb"
                rel="noreferrer noopener"
                target="_blank"
              >
                <img
                  alt="social icon"
                  src="github.svg"
                />
              </a>
            </li>
            <li>
              <a
                href="https://www.linkedin.com/in/daniel-eze-bae"
                rel="noreferrer noopener"
                target="_blank"
              >
                <img
                  alt="social icon"
                  src="linkedin.svg"
                />
              </a>
            </li>
            <li>
              <a
                href="mailto:danibae91@gmail.com"
                rel="noreferrer noopener"
                target="_blank"
              >
                <img
                  alt="social icon"
                  src="at.svg"
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
